ads/improvedigital.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {writeScript, checkData, validateDataExists} from '../src/3p';
+
+const adImproveField = ['width', 'height', 'placement', 'optin', 'keyvalue'];
+const adImproveMandatoryField = ['placement'];
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function improvedigital(global, data)
+{
+  checkData(data, adImproveField);
+  validateDataExists(data, adImproveMandatoryField);
+
+  let url = 'https://ad.360yield.com' +
+    '/adj?' +
+    'p=' + encodeURIComponent(data.placement) +
+    '&w=' + encodeURIComponent(data.width) +
+    '&h=' + encodeURIComponent(data.height) +
+    '&optin=' + encodeURIComponent(data.optin) +
+    '&tz=' + new Date().getTimezoneOffset();
+
+  if (typeof data.keyvalue !== 'undefined' && data.keyvalue) {
+    url += '&' + encodeURIComponent(data.keyvalue);
```

Comment:
Is this really what you want? If `data.keyvalue` is, for instance, "a=b", they will be encoded in your code as `&a%3Db`. If it works for your URL, it's certainly fine, but I wanted to make sure.